Flaky DiskSpaceRecoveryIT.shouldRecoverAfterOutOfDiskSpaceAfterExporting: container-startup timeout

### <!-- Component -->

<!-- Zeebe- -->

### <!-- Affected version -->

<!-- -8.10 -->

### <!-- Severity -->

<!-- Low- -->

### <!-- Likelihood -->

<!-- Low_ -->

### Description

`DiskSpaceRecoveryIT.shouldRecoverAfterOutOfDiskSpaceAfterExporting` (nested `WithStandardContainerTest`, package `io.camunda.zeebe.it.cluster.health`) is intermittently flaky in the `Zeebe Cluster QA ITs` job. The test fails when its Camunda broker test container fails to start within the Testcontainers startup timeout, and passes on retry.

This is an infrastructure/container-startup flake, not a defect in the test logic or product code. In the observed run, three unrelated tests across two classes failed on their first attempt with the identical container-startup timeout and all passed on the immediate retry:

- `io.camunda.zeebe.it.cluster.health.DiskSpaceRecoveryIT.shouldRecoverAfterOutOfDiskSpaceAfterExporting`
- `io.camunda.zeebe.it.cluster.startup.NonDefaultContainerSetupTest.runWithContainerSetup(Consumer)[2]`
- `io.camunda.zeebe.it.cluster.startup.NonDefaultContainerSetupTest.runWithContainerSetup(Consumer)[3]`

Failure signature:

```
org.testcontainers.containers.ContainerLaunchException:
    Container startup failed for image localhost:5000/camunda/camunda:current-test
  Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
  Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
  Caused by: org.rnorth.ducttape.TimeoutException: java.util.concurrent.TimeoutException
```

### Steps to reproduce

1. Run the `Zeebe Cluster QA ITs` integration-test job (`zeebe/qa/integration-tests`).
2. Occasionally the Camunda broker test container exceeds the Testcontainers startup timeout in `@BeforeEach`.
3. `shouldRecoverAfterOutOfDiskSpaceAfterExporting` errors with `ContainerLaunchException` and is reported as a flake (passes on retry).

### Current behavior

The broker test container intermittently fails to start within the Testcontainers timeout, causing the test to error on its first attempt. It passes on the automatic Maven retry (`Run 1` fail → `Run 2` PASS), so the job is reported green with `Flakes: 3`, but the flaky-test gate flags the method.

### Expected behavior

The test should start its broker container reliably and pass on the first attempt, without depending on retries.

### Environment

SM

### Version

- Component Version: 8.10.0-SNAPSHOT (main, commit f7220b1)

### Rootcause

The broker container is created and started, and its `/ready` endpoint responds, but the single partition does not reach a healthy state within the `ZeebeTopologyWaitStrategy` window (`Waiting for PT1M for topology to have ... a leader`). `RaftPartition-1` becomes `HEALTHY`, but `Partition-1` / `ZeebePartitionHealth-1` stays `UNHEALTHY` — first `Services not installed`, then `Snapshot not taken yet` — so the topology wait strategy times out and Testcontainers aborts startup with `ContainerLaunchException`. On the retry the broker has more headroom, becomes healthy in time, and the test passes (`Run 1` fail -> `Run 2` PASS).

This is a slow-startup / resource-contention timing flake on the CI runner (Docker host reported only ~64 GB total memory while running 10 forks in parallel), not a defect in the test or product code. The same run had two other independent container-backed tests (`NonDefaultContainerSetupTest[2]`/`[3]`) fail and recover identically, reinforcing the environmental cause. Container/broker logs from the failed attempt are attached in a comment below.

### Links

- Observed run: https://github.com/camunda/camunda/actions/runs/27746181398/job/82085001726 (commit f7220b1, surfaced in PR #55445)
- Prior (closed) occurrences of the same test: #20003, #11538, #16550

## Container / broker logs from the failed attempt

Source: artifact **Test results for [IT] Zeebe Cluster QA (1)** of run [27746181398](https://github.com/camunda/camunda/actions/runs/27746181398/job/82085001726) — `zeebe/qa/integration-tests/target/failsafe-reports/`.

### Failure (failsafe FLAKY report — `Run 1`, then `Run 2: PASS`)

```
io.camunda.zeebe.it.cluster.health.DiskSpaceRecoveryIT$WithStandardContainerTest
  shouldRecoverAfterOutOfDiskSpaceAfterExporting  (time=27.627s)

org.testcontainers.containers.ContainerLaunchException: Container startup failed for image localhost:5000/camunda/camunda:current-test
  at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:346)
  at io.camunda.container.CamundaContainer.start(CamundaContainer.java:68)
  at org.testcontainers.junit.jupiter.TestcontainersExtension.beforeEach(TestcontainersExtension.java:101)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
  at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
Caused by: org.rnorth.ducttape.TimeoutException: java.util.concurrent.TimeoutException
```

### Testcontainers startup log — `DiskSpaceRecoveryIT-1-output.txt` (the failed fork)

The container starts and `/ready` is polled, but the topology wait strategy never sees a healthy leader:

```
08:25:57.977 tc.localhost:5000/camunda/camunda:current-test - Creating container for image: localhost:5000/camunda/camunda:current-test
08:25:58.390 tc.localhost:5000/camunda/camunda:current-test - Container ... is starting: be49010f5f85...
08:26:49.032 HttpWaitStrategy - /exciting_vaughan: Waiting for 60 seconds for URL: http://localhost:32791/ready (container port 9600)
08:26:49.760 ZeebeTopologyWaitStrategy - /exciting_vaughan: Waiting for PT1M for topology to have at least 1 brokers, 1 partitions with 1 replicas, and each partition to have a leader
08:26:52.888 ZeebeTopologyWaitStrategy - Topology: brokers=[BrokerInfoImpl{nodeId=0, ... partitions=[PartitionInfoImpl{partitionId=1, role=LEADER, health=UNHEALTHY}]}]
```

### Broker health reports — partition never becomes healthy in time

`RaftPartition-1` is `HEALTHY`, but the Zeebe partition services lag (`Services not installed`, then `Snapshot not taken yet`), keeping `Partition-1` `UNHEALTHY` past the wait window:

```
[08:28:14.525] HealthCheckService - Partition-1 failed, marking it as unhealthy:
  HealthReport[componentName=Partition-1, status=UNHEALTHY,
    issue=HealthIssue[message=Services not installed, ...],
    children={ ZeebePartitionHealth-1=UNHEALTHY (Services not installed),
               RaftPartition-1=HEALTHY }]

[08:29:24.199] ZeebePartition-1 - MigrationSnapshotDirector failed, marking it as unhealthy:
  HealthReport[componentName=MigrationSnapshotDirector, status=UNHEALTHY,
    issue=HealthIssue[message=Snapshot not taken yet, retryCount=1, will retry in PT0S, ...]]
```

Docker host in this run reported `Total Memory: 64304 MB` with `forkCount=10` parallel forks — consistent with a resource-contention slow start rather than a code fault.

> Note: GitHub doesn't allow file attachments via API/CLI, so the relevant excerpts are inlined above. The full per-fork outputs (`DiskSpaceRecoveryIT-1/2/3-output.txt`, `NonDefaultContainerSetupTest-1/2-output.txt`) and failsafe XMLs are in the linked run's "Test results for [IT] Zeebe Cluster QA (1)" artifact (not expired).

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/cluster/health/DiskSpaceRecoveryIT.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/cluster/health/DiskSpaceRecoveryIT.java
@@ -98,6 +98,7 @@ final class DiskSpaceRecoveryIT {
                   cfg.getCluster().getNetwork().setMaxMessageSize(DataSize.ofMegabytes(1));
                   cfg.getData().getSecondaryStorage().setType(SecondaryStorageType.none);
                 })
+            .withStartupTimeout(Duration.ofSeconds(90))
             .withEnv("ZEEBE_LOG_LEVEL", "DEBUG")
             .withEnv(UNPROTECTED_API_ENV_VAR, "true")
             .withEnv(AUTHORIZATION_CHECKS_ENV_VAR, "false");

--- a/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/cluster/health/DiskSpaceRecoveryIT.java
+++ b/zeebe/qa/integration-tests/src/test/java/io/camunda/zeebe/it/cluster/health/DiskSpaceRecoveryIT.java
@@ -98,6 +98,7 @@ final class DiskSpaceRecoveryIT {
                   cfg.getCluster().getNetwork().setMaxMessageSize(DataSize.ofMegabytes(1));
                   cfg.getData().getSecondaryStorage().setType(SecondaryStorageType.none);
                 })
+            // Increase startup timeout to reduce CI slow-start flakes; see #55457.
             .withStartupTimeout(Duration.ofSeconds(90))
             .withEnv("ZEEBE_LOG_LEVEL", "DEBUG")
             .withEnv(UNPROTECTED_API_ENV_VAR, "true")

